tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -252,17 +267,18 @@ def test_write_after_pod_restart(self):
                                                     log_prefix=self.test_prefix, skipcleanup=True)
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Step 1: Performed WRITEs/READs/Verify with variable sizes objects.")
-        LOGGER.info("Step 2: Shutdown random data pod by deleting deployment and "
+        LOGGER.info("Step 2: Shutdown random data pod with replica method and "
                     "verify cluster & remaining pods status")
+        num_replica = self.num_replica - 1
         resp = self.ha_obj.delete_kpod_with_shutdown_methods(
             master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
-            down_method=const.RESTORE_DEPLOYMENT_K8S)
+            delete_pod=[self.delete_pod], num_replica=num_replica)
         # Assert if empty dictionary
         assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
         pod_name = list(resp[1].keys())[0]
-        self.deployment_name = resp[1][pod_name]['deployment_name']
-        self.deployment_backup = resp[1][pod_name]['deployment_backup']
+        self.set_name = resp[1][pod_name]['deployment_name']
         self.restore_method = resp[1][pod_name]['method']
+        pod_name = list(resp[1].keys())[0]
```

Comment:
Statefulset changes are needed for restarting pod also, please check at all the places.